OpenRA.Game/Platform.cs

```diff
@@ -66,111 +74,100 @@ public static string RuntimeVersion
+					systemSupportPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), "OpenRA") + Path.DirectorySeparatorChar;
 					break;
 				}
 
 				case PlatformType.OSX:
 				{
-					preferredSupportDir = fallbackSupportDir = Path.Combine(
+					modernUserSupportPath = legacyUserSupportPath = Path.Combine(
 						Environment.GetFolderPath(Environment.SpecialFolder.Personal),
 						"Library", "Application Support", "OpenRA");
+
+					systemSupportPath = "/Library/Application Support/OpenRA/";
 					break;
 				}
 
 				case PlatformType.Linux:
 				{
-					fallbackSupportDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".openra");
+					legacyUserSupportPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".openra");
 
 					var xdgConfigHome = Environment.GetEnvironmentVariable("XDG_CONFIG_HOME");
 					if (string.IsNullOrEmpty(xdgConfigHome))
 						xdgConfigHome = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".config");
 
-					preferredSupportDir = Path.Combine(xdgConfigHome, "openra");
+					modernUserSupportPath = Path.Combine(xdgConfigHome, "openra");
+					systemSupportPath = "/var/games/openra/";
 
 					break;
 				}
 
 				default:
 				{
-					preferredSupportDir = fallbackSupportDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".openra");
+					modernUserSupportPath = legacyUserSupportPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Personal), ".openra");
+					systemSupportPath = "/var/games/openra/";
 					break;
 				}
 			}
 
+			// Use a local directory in the game root if it exists (shared with the system support dir)
+			var localSupportDir = Path.Combine(GameDir, "Support");
+			if (Directory.Exists(localSupportDir))
+				userSupportPath = systemSupportPath = localSupportDir + Path.DirectorySeparatorChar;
+
 			// Use the fallback directory if it exists and the preferred one does not
-			if (!Directory.Exists(preferredSupportDir) && Directory.Exists(fallbackSupportDir))
-				return fallbackSupportDir + Path.DirectorySeparatorChar;
+			else if (!Directory.Exists(modernUserSupportPath) && Directory.Exists(legacyUserSupportPath))
+				userSupportPath = legacyUserSupportPath + Path.DirectorySeparatorChar;
+			else
+				userSupportPath = modernUserSupportPath + Path.DirectorySeparatorChar;
 
-			return preferredSupportDir + Path.DirectorySeparatorChar;
+			supportDirInitialized = true;
 		}
 
 		/// <summary>
-		/// Directory containing system-wide support files (mod metadata).
-		/// This directory is not guaranteed to exist or be writable.
-		/// Consumers are expected to check the validity of the returned value, and
-		/// fall back to the user support directory if necessary.
+		/// Specify a custom support directory that already exists on the filesystem.
+		/// MUST be called before Platform.SupportDir is first accessed.
```

Comment:
The issue is that we don't know what callers will be doing with the support dir value - many of them will combine it with some other path components and then cache the result without ever reading the new value. This would lead to all sorts of awful to debug issues.

The first thing this method does is check whether the value has been initialized (i.e. `GetSupportDir` has been called) and throw an exception if it has.